Public/Src/Cache/ContentStore/Library/Service/Grpc/BufferedReadStream.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace BuildXL.Cache.ContentStore.Service.Grpc
+{
+
+    // This class is required for decompressing data streamed via grpc
+    // because the compression api insists on reading from a stream and
+    // the grpc api exposes a read-one-chunk method instead of a stream.
+
+    internal class BufferedReadStream : Stream

```

Comment:
It seems that the logic is relatively involved, so maybe it make sense to create at least a few tests to cover the basic functionality.